modules/javafx.base/src/test/java/test/util/memory/JMemoryBuddy.java

```diff
@@ -34,31 +34,43 @@
 import java.text.SimpleDateFormat;
 import java.util.Date;
 import java.util.LinkedList;
+import java.util.Objects;
 import java.util.function.Consumer;
 import java.util.function.Function;
 
 
 /**
+ * JMemoryBuddy provides various methods to test for memory leaks.
+ * It makes it easy to verify the memory behavior in a unit test ensuring the stability and quality of your code.
  * Checkout <a href="https://github.com/Sandec/JMemoryBuddy">https://github.com/Sandec/JMemoryBuddy</a> for more documentation.
  */
 public class JMemoryBuddy {
 
     private static int steps = 10;
-    private static int overallTime = 1000;
-    private static int sleepTime = overallTime / steps;
+    private static int testDuration = 1000;
+    private static int sleepDuration = testDuration / steps;
     private static boolean createHeapdump = false;
     private static int garbageAmount = 999999;
-    private static String MX_BEAN_PROXY_TYPE = "com.sun.management:type=HotSpotDiagnostic";
+    private static String mxBeanProxyName = "com.sun.management:type=HotSpotDiagnostic";
     private static String outputFolderString = ".";
 
     static {
-        outputFolderString = System.getProperty("jmemorybuddy.output",".");
-        overallTime = Integer.parseInt(System.getProperty("jmemorybuddy.checktime","1000"));
+        outputFolderString = System.getProperty("jmemorybuddy.output", getDefaultOutputFolder());
+        testDuration = Integer.parseInt(System.getProperty("jmemorybuddy.testDuration","1000"));
         steps = Integer.parseInt(System.getProperty("jmemorybuddy.steps", "10"));
         createHeapdump = Boolean.parseBoolean(System.getProperty("jmemorybuddy.createHeapdump", "false"));
         garbageAmount = Integer.parseInt(System.getProperty("jmemorybuddy.garbageAmount", "10"));
     }
 
+    private static String getDefaultOutputFolder() {
+        File folder1 = new File("target");
+        File folder2 = new File("build");
+
+        if(folder1.exists()) return folder1.getAbsolutePath();
+        if(folder2.exists()) return folder2.getAbsolutePath();
```

Comment:
Space after `if` (I see you corrected most of them elsewhere in the code, thanks).

Reply:
done